File: 241114/선분 3개 지우기/remove-three-segments.py
n = int(input())
arr = [list(map(int, input().split())) for _ in range(n)]
cnt = 0

for i in range(n):
    for j in range(i+1, n):
        for k in range(j+1, n):
            x = [0]*101
            for m in range(n):
                if m == i or m == j or m == k:
                    continue
                for o in range(arr[m][0], arr[m][1]+1):
                    x[o] += 1
            check = True
            for l in x:
                if l >= 2:
                    check = False
                    break
            if check:
                cnt += 1

print(cnt)

File: 250124/선분 3개 지우기/remove-three-segments.cpp
#include <iostream>
using namespace std;

int main() {
    // Please write your code here.
    int x[10], y[10];

    int n;
    cin >> n;
    for (int i=0; i<n; i++)
        cin >> x[i] >> y[i];

    int cnt = 0;
    for (int i=0; i<n; i++) {
        for (int j=i+1; j<n; j++) {
            for (int k=j+1; k<n; k++) {

                int arr[101] = {};
                for (int l=0; l<n; l++) {
                    if (l == i || l == j || l == k) continue;
                    for (int m=x[l]; m<=y[l]; m++) {
                        arr[m]++;
                    }
                }

                bool check = true;
                for (int l=0; l<=100; l++) {
                    if (arr[l] == 0 || arr[l] == 1) continue;
                    check = false;
                    break;
                }
                
                if (check) cnt++;
            }
        }
    }
    cout << cnt;
    
    return 0;
}